test/metabase/query_processor_test/coercion_test.clj

```diff
@@ -83,6 +83,21 @@
               (is (= res
                      (biginteger coerced-number))))))))))
 
+(defn- date-type? [col]
+  (some #(types/field-is-type? % col) [:type/DateTime ;; some databases return datetimes for date (e.g., Oracle)
+                                       :type/Text ;; sqlite uses text :(
+                                       :type/Date]))
```

Comment:
Does this one not run on mongo, or why doesn't it need `:type/*` like the one above?

Reply:
It does run on Mongo. And it's calling the same function as the other one. But somehow this gets Mongo to return a date!